appium-tests/helpers/contactsHelper.js

```diff
@@ -0,0 +1,222 @@
+
+    var result =
+        'var results = document.getElementById("info");\n' +
+        'var obj = new ContactFindOptions();\n' +
+        'obj.filter = ' + preparedOldDisplayName + ';\n' +
+        'obj.multiple = false;\n' +
+
+        'navigator.contacts.find(["displayName", "name"], function(contacts) {\n' +
+        '    if (contacts.length === 0) {\n' +
+        '        results.innerHTML = "No contacts to update.";\n' +
+        '        return;\n' +
+        '    }\n' +
+        '    var contact = contacts[0];\n' +
+        '    contact.displayName = ' + preparedNewDisplayName + ';\n' +
+        '    contact.name = ' + preparedNewName + ';\n' +
+        '    contact.save(function(updated) {\n' +
+        '        results.innerHTML = "Contact updated.";\n' +
+        '    },function(e) {\n' +
+        '        results.innerHTML = "Update failed: error " + e.code;\n' +
+        '    });\n' +
+        '}, function(e) {\n' +
+        '    results.innerHTML = "Search failed: error " + e.code;\n' +
+        '}, obj);';
+
+    return result;
+};
+
+module.exports.getRemoveTestContactsCode = function () {
+    var result =
+        'var results = document.getElementById("info");\n' +
+        'results.innerHTML = "";\n' +
+        'var obj = new ContactFindOptions();\n' +
+        'obj.filter = "DeleteMe";\n' +
+        'obj.multiple = true;\n' +
+        'navigator.contacts.find(["note"], function(contacts) {\n' +
+        '    var removes = [];\n' +
+        '    contacts.forEach(function(contact) {\n' +
+        '        removes.push(contact);\n' +
+        '    });\n' +
+        '    if (removes.length === 0) {\n' +
+        '        results.innerHTML = "No contacts to remove";\n' +
+        '        return;\n' +
+        '    }\n' +
+
+        '   var nextToRemove;\n' +
+        '   if (removes.length > 0) {\n' +
+        '        nextToRemove = removes.shift();\n' +
+        '    }\n' +
+
+        '    function removeNext(item) {\n' +
+        '        if (typeof item === "undefined") {\n' +
+        '            return;\n' +
+        '        }\n' +
+
+        '        if (removes.length > 0) {\n' +
+        '            nextToRemove = removes.shift();\n' +
+        '        } else {\n' +
+        '            nextToRemove = undefined;\n' +
+        '        }\n' +
+
+        '        item.remove(function removeSucceeded() {\n' +
+        '            results.innerHTML += "Removed a contact with ID " + item.id + "<br/>";\n' +
+        '            removeNext(nextToRemove);\n' +
+        '        }, function removeFailed() {\n' +
+        '            results.innerHTML += "Failed to remove a contact with ID " + item.id + "<br/>";\n' +
+        '            removeNext(nextToRemove);\n' +
+        '        });\n' +
+        '    }\n' +
+        '    removeNext(nextToRemove);\n' +
+        '}, function(e) {\n' +
+        '   results.innerHTML = "Search failed: error " + e.code;\n' +
+        '}, obj);';
+
+    return result;
+};
```

Comment:
This is the code to run on the client side. We inject it by calling `driver.execute()` from the specs.